csp/tests/test_utils.py

```diff
@@ -211,3 +211,8 @@ def test_upgrade_insecure_requests():
 def test_block_all_mixed_content():
     policy = build_policy()
     policy_eq("default-src 'self'; block-all-mixed-content", policy)
+
+
+def test_nonce():
+    policy = build_policy(nonce='abc123')
+    policy_eq("default-src 'self'; nonce-abc123", policy)
```

Comment:
As @moggers87 pointed out, nonces should end up in source lists (all the *-src directives) like `default-src 'self' nonce-abc123`.

A couple ways to do this from old PRs and such:

* lots of keyword params e.g.

```python
>>> build_policy(default_src_nonce='abc123')
"default-src 'self' nonce-abc123"
```

* insert a placeholder in the existing lists e.g.

settings.py

```python
CSP_DEFAULT_SRC = (
     'self',
     'INCLUDE_NONCE_HERE',
)
>>> build_policy()
"default-src 'self' nonce-abc123"
```

Maybe there's a better way too.